tests/CMakeLists.txt

```diff
@@ -30,42 +30,219 @@ set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+# Settings test object library
+add_library(test_settings_obj OBJECT
+  test_global_settings_handlers.cpp
+  test_instance_settings_handler.cpp
+  test_persistent_settings_handler.cpp
+  test_remote_settings_handler.cpp
+  test_setting_specs.cpp
+  test_settings.cpp
+)
+target_include_directories(test_settings_obj PRIVATE ${CMAKE_SOURCE_DIR}/src/settings)
+target_link_libraries(test_settings_obj PRIVATE
+  test_common_deps
+  settings
+)
+
+# Network and parser object library for tests needing protobuf
+add_library(test_network_parser_obj OBJECT
+  test_argparser.cpp
+)
+target_link_libraries(test_network_parser_obj PRIVATE
+  test_common_deps
+  protobuf::libprotobuf
+  client
+  daemon
+)
+
+# Core object libraries
+add_library(test_minimal_obj OBJECT
   common.cpp
-  daemon_test_fixture.cpp
+  main.cpp
+  mock_logger.cpp
+  mock_standard_paths.cpp
+  temp_dir.cpp
+  temp_file.cpp
+)
+target_link_libraries(test_minimal_obj PRIVATE test_common_deps)
+
+add_library(test_fs_obj OBJECT
   file_operations.cpp
-  image_host_remote_count.cpp
+  path.cpp
+)
+target_link_libraries(test_fs_obj PRIVATE test_common_deps)
+
+add_library(test_json_obj OBJECT
   json_test_utils.cpp
-  main.cpp
+  blueprint_test_lambdas.cpp
+)
+target_link_libraries(test_json_obj PRIVATE test_common_deps protobuf::libprotobuf)
+
+add_library(test_network_obj OBJECT
   mischievous_url_downloader.cpp
-  mock_logger.cpp
+  image_host_remote_count.cpp
+)
+target_link_libraries(test_network_obj PRIVATE test_common_deps)
+
+add_library(test_daemon_obj OBJECT
+  daemon_test_fixture.cpp
+)
+target_link_libraries(test_daemon_obj PRIVATE test_common_deps protobuf::libprotobuf daemon)
+
+add_library(test_openssl_obj OBJECT
   mock_openssl_syscalls.cpp
+)
+target_link_libraries(test_openssl_obj PRIVATE test_common_deps)
+
+# Process and SSH files that were causing issues as object libraries
+set(TEST_PROCESS_SOURCES
   mock_process_factory.cpp
-  mock_sftp.cpp
-  mock_sftpserver.cpp
+  stub_process_factory.cpp
+  reset_process_factory.cpp
+)
+
+set(TEST_SSH_SOURCES
   mock_ssh.cpp
   mock_ssh_client.cpp
-  mock_standard_paths.cpp
-  path.cpp
-  reset_process_factory.cpp
-  stub_process_factory.cpp
-  temp_dir.cpp
-  temp_file.cpp
-  test_alias_dict.cpp
-  test_argparser.cpp
-  test_base_snapshot.cpp
-  test_base_virtual_machine.cpp
-  test_base_virtual_machine_factory.cpp
-  test_basic_process.cpp
+  mock_sftp.cpp
+  mock_sftpserver.cpp
+)
+
+# Common source combinations using object libraries
+set(MINIMAL_TEST_OBJECTS
+  $<TARGET_OBJECTS:test_minimal_obj>
+  $<TARGET_OBJECTS:test_openssl_obj>
+)
+
+set(NETWORK_TEST_OBJECTS
+  ${MINIMAL_TEST_OBJECTS}
+  $<TARGET_OBJECTS:test_network_obj>
+  $<TARGET_OBJECTS:test_fs_obj>
+)
+
+set(SSH_CLIENT_TEST_OBJECTS
+  ${MINIMAL_TEST_OBJECTS}
+  ${TEST_PROCESS_SOURCES}
+  $<TARGET_OBJECTS:test_fs_obj>
+  ${TEST_SSH_SOURCES}
+  $<TARGET_OBJECTS:test_json_obj>
+  $<TARGET_OBJECTS:test_daemon_obj>
+)
+
+set(DAEMON_TEST_OBJECTS
+  ${MINIMAL_TEST_OBJECTS}
```

Comment:
I feel like we're trying to solve a problem which is already solved before. $<TARGET_OBJECTS:obj_lib_name> is effectively a static library, except that a static library also handles the linking. Moreover, the object libraries can be directly linked against and will have the same effect as expanding the objects into the source part.